apis/core/v1alpha1/condition.go

```diff
@@ -53,6 +56,13 @@ const (
 	ReasonReconcileError   ConditionReason = "Encountered an error during managed resource reconciliation"
 )
 
+// Reason references for a resource are or are not resolved
+const (
+	ReasonReferenceResolveSuccess  ConditionReason = "Successfully resolved managed resource references to other resources"
```

Comment:
@muvaf thanks for the review!

I think my main goal here was to make the cross referencing as _managed_ as possible, by laying out the overall required steps, and detecting the references automatically, rather than leaving it to the developer to write the logic and then just simply execute it by putting it in `SetupWithManager `. Taking this slightly more managed approach, has the following advantages:

- More readability, and visibility through using struct field tags
- Enforcing how resolution steps should look like
- Early panic if a referencer is not implemented correctly

That said, I agree that there is a trade-off here. It would be possible to go the `Initializer` route and avoid the reflection logic, but then the mentioned goodies won't be there. Given that I am already invested in this implementation, I prefer keeping the existing approach.